packages/material-renderers/src/controls/MaterialDateControl.tsx

```diff
@@ -73,17 +80,28 @@ export const MaterialDateControl = (props: ControlProps) => {
     ? errors
     : null;
   const secondFormHelperText = showDescription && !isValid ? errors : null;
+
+  const updateChild = () => {
+    setKey(key + 1);
+  };
```

Comment:
Whenever the new value of a state depends on the old value, the functional updater should be used, i.e. `const updateChild = () => setKey((key) => key + 1)`